packages/protocol/chainmanager/manager.go

```diff
@@ -3,24 +3,36 @@ package chainmanager
 import (
 	"sync"
 
+	"github.com/pkg/errors"
+
 	"github.com/iotaledger/hive.go/core/generics/event"
 	"github.com/iotaledger/hive.go/core/generics/options"
 	"github.com/iotaledger/hive.go/core/generics/walker"
+	"github.com/iotaledger/hive.go/core/identity"
 	"github.com/iotaledger/hive.go/core/syncutils"
-	"github.com/pkg/errors"
 
 	"github.com/iotaledger/goshimmer/packages/core/commitment"
+	"github.com/iotaledger/goshimmer/packages/core/epoch"
 	"github.com/iotaledger/goshimmer/packages/core/eventticker"
+	"github.com/iotaledger/goshimmer/packages/core/memstorage"
+)
+
+var (
+	ErrCommitmentUnknown  = errors.New("unknown commitment")
+	ErrCommitmentNotSolid = errors.New("commitment not solid")
 )
 
 type Manager struct {
 	Events              *Events
 	SnapshotCommitment  *Commitment
 	CommitmentRequester *eventticker.EventTicker[commitment.ID]
 
+	//TODO: change to memstorage.EpochStorage?
 	commitmentsByID      map[commitment.ID]*Commitment
 	commitmentsByIDMutex sync.Mutex
 
+	commitmentsByForkingPoint *memstorage.EpochStorage[commitment.ID, *ForkDetectedEvent]
```

Comment:
Seems to be a bit odd to store an event here. Maybe we can just rename this struct to something else?